superset/migrations/shared/catalogs.py

```diff
@@ -57,17 +126,41 @@ def upgrade_schema_perms(engine: str | None = None) -> None:
                 None,
                 schema,
             )
-            existing_pvm = security_manager.find_permission_view_menu(
-                "schema_access",
-                perm,
-            )
+            existing_pvm = session.query(ViewMenu).filter_by(name=perm).one_or_none()
             if existing_pvm:
-                existing_pvm.view_menu.name = security_manager.get_schema_perm(
+                existing_pvm.name = security_manager.get_schema_perm(
                     database.database_name,
                     catalog,
                     schema,
                 )
 
+        # update existing models
+        models = [
+            (Query, "database_id"),
+            (SavedQuery, "db_id"),
+            (TabState, "database_id"),
+            (TableSchema, "database_id"),
+            (SqlaTable, "database_id"),
+        ]
```

Comment:
do we have tests for the `upgrade_schema_perms` and `downgrade_schema_perms`?

Reply:
We don't, let me add them!

Reply:
Added.